packages/angular/common/src/directives/navigation/tabs.ts

```diff
@@ -142,7 +178,45 @@ export abstract class IonTabs implements AfterContentInit, AfterContentChecked {
     }
   }
 
+  private setActiveTab(tab: string): void {
+    const tabs = this.tabs;
+    const selectedTab = tabs.find((t: any) => t.tab === tab);
+
+    if (!selectedTab) {
+      console.error(`tab with id: "${tab}" does not exist`);
+      return;
+    }
+
+    this.leavingTab = this.selectedTab;
+    this.selectedTab = selectedTab;
+
+    this.ionTabsWillChange.emit({ tab });
+
+    selectedTab.el.active = true;
+  }
+
+  private tabSwitch(): void {
+    const selectedTab = this.selectedTab;
```

Comment:
~ These two lines could be minimized to:
```ts
const { selectedTab, leavingTab } = this;
```

Reply:
https://github.com/ionic-team/ionic-framework/pull/29786/commits/81193e1fc6e93e6eb682a9d963eb6526c44cd2aa